packages/tse-backend/src/modules/exchange-trade/model/exchange-trade.model.ts

```diff
@@ -0,0 +1,101 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { AutoMap } from '@automapper/classes';
+import { IsString, IsNumber, IsEnum } from 'class-validator';
+import { TradeSideType } from '../../../common/enums/exchange-operation.enums';
+
+export class MarketTradeDto {
+  @AutoMap()
+  @ApiProperty()
+  @IsString()
+  userId: string;
+
+  @AutoMap()
+  @ApiProperty()
+  @IsString()
+  clientId: string;
+
+  @AutoMap()
+  @ApiProperty()
+  @IsString()
+  exchange: string;
+
+  @AutoMap()
+  @ApiProperty({ example: 'BTC/USDT' })
+  @IsString()
+  symbol: string;
+
+  @AutoMap()
+  @ApiProperty({
+    example: 'buy',
+    enum: TradeSideType,
+  })
+  @IsEnum(TradeSideType)
+  side: TradeSideType;
+
+  @AutoMap()
+  @ApiProperty()
+  @IsNumber()
+  amount: number;
+}
+
+export class MarketLimitDto extends MarketTradeDto {
+  @AutoMap()
+  @ApiProperty()
+  @IsNumber()
+  price: number;
+}
+
+export class MarketTradeCommand {
+  @AutoMap()
+  userId: string;
+
+  @AutoMap()
+  clientId: string;
+
+  @AutoMap()
+  exchange: string;
+
+  @AutoMap()
+  symbol: string;
+
+  @AutoMap()
+  side: string;
+
+  @AutoMap()
+  amount: number;
+}
+
+export class MarketLimitCommand extends MarketTradeCommand {
+  @AutoMap()
+  price: number;
+}
+
+export type MarketTrade = MarketLimitCommand | MarketTradeCommand;
+
+export class CancelOrderDto {
+  @AutoMap()
+  @ApiProperty()
+  @IsString()
+  exchange: string;
+
+  @AutoMap()
+  @ApiProperty()
+  @IsString()
+  orderId: string;
+
+  @AutoMap()
+  @ApiProperty()
+  @IsString()
+  symbol: string;
+}
+
+export class CancelOrderCommand {
```

Comment:
Use `interface` instead of `class` if you want to create a wrapper:
https://stackoverflow.com/questions/51716808/when-use-a-interface-or-class-in-typescript

and try not to extend classes, "Favour Composition over Inheritance:
https://en.wikipedia.org/wiki/Composition_over_inheritance"

Reply:
it's for all wrappers

Reply:
In NestJS for DTO we need to use classes to get the benefits of validation, transformation and documentation features.
For Commands decorator @AutoMap() doesn’t work with interfaces..